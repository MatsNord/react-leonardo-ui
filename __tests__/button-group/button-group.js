import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../../src/button';
import ButtonGroup from '../../src/button-group';

test('Empty ButtonGroup renders correctly', () => {
  const tree = renderer.create(
    <ButtonGroup />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('ButtonGroup with buttons renders correctly', () => {
  const tree = renderer.create(
    <ButtonGroup>
      <Button>Button 1</Button>
      <Button>Button 2</Button>
      <Button variant="info">Button 3</Button>
    </ButtonGroup>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Inverse ButtonGroup with buttons renders correctly', () => {
  const tree = renderer.create(
    <ButtonGroup variant="inverse">
      <Button variant="inverse">Button 1</Button>
      <Button variant="inverse">Button 2</Button>
      <Button variant="info">Button 3</Button>
    </ButtonGroup>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('ButtonGroup passes props correctly', () => {
  const onClick = jest.fn(() => {});

  const tree = renderer.create(
    <ButtonGroup className="my-class" data-x="x" title="Title" onClick={onClick} />
  ).toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onClick();
  expect(onClick.mock.calls.length).toBe(1);
});
