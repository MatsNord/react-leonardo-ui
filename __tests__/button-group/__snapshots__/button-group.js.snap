exports[`test ButtonGroup passes props correctly 1`] = `
<div
  className="lui-buttongroup my-class"
  data-x="x"
  onClick={[Function]}
  title="Title" />
`;

exports[`test ButtonGroup with buttons renders correctly 1`] = `
<div
  className="lui-buttongroup">
  <button
    className="lui-button lui-buttongroup__button  ">
    Button 1
  </button>
  <button
    className="lui-button lui-buttongroup__button  ">
    Button 2
  </button>
  <button
    className="lui-button lui-button--info lui-buttongroup__button  ">
    Button 3
  </button>
</div>
`;

exports[`test Empty ButtonGroup renders correctly 1`] = `
<div
  className="lui-buttongroup" />
`;

exports[`test Inverse ButtonGroup with buttons renders correctly 1`] = `
<div
  className="lui-buttongroup"
  variant="inverse">
  <button
    className="lui-button lui-button--inverse lui-buttongroup__button  ">
    Button 1
  </button>
  <button
    className="lui-button lui-button--inverse lui-buttongroup__button  ">
    Button 2
  </button>
  <button
    className="lui-button lui-button--info lui-buttongroup__button  ">
    Button 3
  </button>
</div>
`;
